src/query/v2/plan/vertex_count_cache.hpp

```diff
@@ -52,11 +55,16 @@ class VertexCountCache {
     return 1;
   }
 
-  // For now return true if label is primary label
-  bool LabelIndexExists(storage::v3::LabelId label) { return request_router_->IsPrimaryLabel(label); }
+  bool LabelIndexExists(storage::v3::LabelId label) { return PrimaryLabelExists(label); }
```

Comment:
This used to be the original functionality of this function, this should not throw, since every time an scan operator of the `ScanAllByLabel` would be initiated, it would fail, instead of picking another operator this broke some tests previously. Returning false, breaks some of the e2e tests, so for now, until we have proper indexing based on labels, I would opt for this implementation.